dhis-2/dhis-services/dhis-service-validation/src/main/java/org/hisp/dhis/validation/DefaultValidationResultService.java

```diff
@@ -106,16 +107,29 @@ public void saveValidationResults( Collection<ValidationResult> validationResult
         return validationResultStore.getAllUnreportedValidationResults();
     }
 
+    @Transactional( readOnly = false )
     @Override
     public void deleteValidationResult( ValidationResult validationResult )
     {
         validationResultStore.delete( validationResult );
     }
 
+    @Transactional( readOnly = false )
```

Comment:
We have a `readOnly=true` on methods convention, i.e. no @Transactional at class level, just on method level. No value for read-write and then `readOnly=true` for read-only.